connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -582,12 +583,20 @@ void incrementalAlterConfigs(Map<String, Config> topicConfigs) {
     }
 
     private void updateTopicAcls(List<AclBinding> bindings) {
-        log.trace("Syncing {} topic ACL bindings.", bindings.size());
-        targetAdminClient.createAcls(bindings).values().forEach((k, v) -> v.whenComplete((x, e) -> {
-            if (e != null) {
-                log.warn("Could not sync ACL of topic {}.", k.pattern().name(), e);
-            }
-        }));
+        Set<AclBinding> addBindings = new HashSet<>();
+        addBindings.addAll(bindings);
+        addBindings.removeAll(knownTopicAclBindings);
```

Comment:
My IDE nagged me about possible slow performance for invoking `Set::removeAll` with a `List` as an argument. Some research led to [this fascinating blog post](https://codeblog.jonskeet.uk/2010/07/29/there-s-a-hole-in-my-abstraction-dear-liza-dear-liza/).

I think the scenario described there isn't likely to impact us frequently, but just in case, do you think we can change `knownTopicAclBindings` from a `List` to a `Set`?

Reply:
Thank you very much. I learned from this blog. 
I have changed `knownTopicAclBindings` from `List` to `Set`. 
I would like to ask a digression: which is your IDE? 
What plug-in tool prompts slow performance? @C0urante